Detectors/ZDC/raw/src/DumpRaw.cxx

```diff
@@ -0,0 +1,247 @@
+    }
+  }
+  for (UInt_t i = 0; i < NDigiChannels; i++) {
+    if (mCounts[i] && mCounts[i]->GetEntries() > 0) {
+      setStat(mCounts[i]);
+      mCounts[i]->Write();
+    }
+  }
+  for (UInt_t i = 0; i < NDigiChannels; i++) {
+    if (mSignal[i] && mSignal[i]->GetEntries() > 0) {
+      setStat(mSignal[i]);
+      mSignal[i]->Write();
+    }
+  }
+  f->Close();
+}
+
+inline int DumpRaw::getHPos(uint32_t board, uint32_t ch)
+{
+  int ih = board * 4 + ch;
+  if (ih < NDigiChannels) {
+    return ih;
+  } else {
+    LOG(ERROR) << "Wrong ih " << ih << " board " << board << " ch " << ch;
+    return -1;
+  }
+}
+
+int DumpRaw::processWord(const UInt_t* word)
+{
+  if (word == 0) {
+    printf("NULL\n");
+    return 1;
+  }
+  if ((word[0] & 0x3) == Id_w0) {
+    for (Int_t iw = 0; iw < NWPerGBTW; iw++) {
+      mCh.w[0][iw] = word[iw];
+    }
+  } else if ((word[0] & 0x3) == Id_w1) {
+    if (mCh.f.fixed_0 == Id_w0) {
+      for (Int_t iw = 0; iw < NWPerGBTW; iw++) {
+        mCh.w[1][iw] = word[iw];
+      }
+    } else {
+      LOG(ERROR) << "Wrong word sequence";
+      mCh.f.fixed_0 = Id_wn;
+      mCh.f.fixed_1 = Id_wn;
+      mCh.f.fixed_2 = Id_wn;
+    }
+  } else if ((word[0] & 0x3) == Id_w2) {
+    if (mCh.f.fixed_0 == Id_w0 && mCh.f.fixed_1 == Id_w1) {
+      for (Int_t iw = 0; iw < NWPerGBTW; iw++) {
+        mCh.w[2][iw] = word[iw];
+      }
+      process(mCh);
+    } else {
+      LOG(ERROR) << "Wrong word sequence";
+    }
+    mCh.f.fixed_0 = Id_wn;
+    mCh.f.fixed_1 = Id_wn;
+    mCh.f.fixed_2 = Id_wn;
+  } else {
+    // Word not present in payload
+    LOG(FATAL) << "Event format error";
+    return 1;
+  }
+  return 0;
+}
+
+int DumpRaw::process(const EventChData ch)
+{
+  // Not empty event
+  auto f = ch.f;
+  int ih = getHPos(f.board, f.ch);
+  if (mVerbosity > 0) {
+    for (Int_t iw = 0; iw < NWPerBc; iw++) {
+      Digits2Raw::print_gbt_word(ch.w[iw]);
+    }
+  }
+  UShort_t us[12];
+  Short_t s[12];
+  us[0] = f.s00;
+  us[1] = f.s01;
+  us[2] = f.s02;
+  us[3] = f.s03;
+  us[4] = f.s04;
+  us[5] = f.s05;
+  us[6] = f.s06;
+  us[7] = f.s07;
+  us[8] = f.s08;
+  us[9] = f.s09;
+  us[10] = f.s10;
+  us[11] = f.s11;
+  for (Int_t i = 0; i < 12; i++) {
+    if (us[i] > ADCMax) {
+      s[i] = us[i] - ADCRange;
+    } else {
+      s[i] = us[i];
+    }
+    //printf("%d %u %d\n",i,us[i],s[i]);
+  }
+  if (f.Alice_3) {
+    for (Int_t i = 0; i < 12; i++)
+      mSignal[ih]->Fill(i - 36., Double_t(s[i]));
+  }
+  if (f.Alice_2) {
+    for (Int_t i = 0; i < 12; i++)
+      mSignal[ih]->Fill(i - 24., Double_t(s[i]));
+  }
+  if (f.Alice_1 || f.Auto_1) {
+    for (Int_t i = 0; i < 12; i++)
+      mSignal[ih]->Fill(i - 12., Double_t(s[i]));
+  }
+  if (f.Alice_0 || f.Auto_0) {
+    for (Int_t i = 0; i < 12; i++)
+      mSignal[ih]->Fill(i + 0., Double_t(s[i]));
+    Double_t bc_d = UInt_t(f.bc / 100);
+    Double_t bc_m = UInt_t(f.bc % 100);
+    mBunch[ih]->Fill(bc_m, -bc_d);
+  }
+  if (f.bc == 3563) {
```

Comment:
better use predefined constants wherever possible:
``if (f.bc == o2::constants::lhc::LHCMaxBunches-1) {``
(will need to ``#include "CommonConstants/LHCConstants.h"``)